### ISSUE TITLE
[Feature Request] Add login on page

### ISSUE THREAD

**Issue by HomerusJa** [dmotts/yt-to-cloud-uploader#28]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Feature Description

To make the app more usable, we should add the feature to log in directly on the server without modifying a file on the serverside.

### Use Case

It would enhance the usability.

### Priority

Medium

### Record

- [X] I have read the Contributing Guidelines
- [X] I have starred the repository

**Comment by HomerusJa** [comment]
This is different to #17 because it right here, you should be able to input your keys and stuff. I think, this should be done before proceeding with #17.

**Comment by HomerusJa** [comment]
Should we directly go with a solution which would allow multiple passwords and so on and which stores them in a SQLite database? This would be significantly more work, but we wouldn't just redo this issue again when we start working on #17.

**Comment by dmotts** [comment]
Multiple passwords for a single user. @HomerusJa please elaborate if you will.

**Comment by dmotts** [comment]
@HomerusJa at this stage, I believe we should keep things fairly simple not to create unnecessary complexity. However, since you restructured the application. I'll let you decide what you think is best. Remember, this is suppose to be fun, we're in it for the love of code! Have fun with it! 😃

**Comment by shaikmynuddin** [comment]
Hi @dmotts can I help
looking forward for your reply

**Comment by dmotts** [comment]
Hey @shaikmynuddin, this task has been assigned to @HomerusJa so I'll let him decide if he'll want a hand. In the meantime, feel free to explore the other [issues](https://github.com/dmotts/yt-to-cloud-uploader/issues) or [open source projects](https://github.com/stars/dmotts/lists/open-source-projects) available.

**Comment by HomerusJa** [comment]
@shaikmynuddin You can happily work on this. I’ll be gone for the next few days, so there you go!

**Comment by HomerusJa** [comment]
You might want to have a look at this: https://github.com/mkhorasani/Streamlit-Authenticator , but I think this can also easily be implemented by yourself.

**Comment by dmotts** [comment]
@HomerusJa @shaikmynuddin @swarupn17  @shawaiz-khan Guys please note, that we are moving away from streamlit to a more suitable option. I implore you to express your thoughts an ideas here -> #31

**Comment by dmotts** [comment]
Thank you for your co-operation guys! 💪💯🙌

**Comment by dmotts** [comment]
If I didn't mention your name, it just means it didn't show up on my dropdown menu.
Just know, when I say 'guys' or 'people' or 'team'. I talking to everyone! 😄

**Comment by dmotts** [comment]
Let's go! 💥💯,💥

**Comment by shaikmynuddin** [comment]
> @HomerusJa @shaikmynuddin @swarupn17 @shawaiz-khan Guys please note, that we are moving away from streamlit to a more suitable option. I implore you to express your thoughts and ideas here -> #31

Yes I am looking into that also first I will complete these as some Example login page

**Comment by dmotts** [comment]
Great @shaikmynuddin! I'm looking forward to seeing it. 😁

**Comment by dmotts** [comment]
Hey @shaikmynuddin. This may require the frontend to be completed however, creating the API would be good to go since the project has a new structure.

**Comment by shaikmynuddin** [comment]
> Owner

I have worked on authentication I used flask and react for that simple sign up and login first we have change the streamlit content to web then I can integrate it features

**Comment by shaikmynuddin** [comment]
![Screenshot 2024-10-20 090338](https://github.com/user-attachments/assets/8da73e10-cfc8-46e0-a133-573d90de06e7)
code
app = Flask(__name__)
CORS(app, supports_credentials=True)
app.config.from_object(ApplicationConfig)
db.init_app(app)
bcrypt = Bcrypt(app)
Session(app)
with app.app_context():
    db.create_all()

@app.route("/@me")
def get_current_user():
    user_id = session.get("user_id")

    if not user_id:
        return jsonify({"error": "Unauthorized"}), 401
    
    user = User.query.filter_by(id=user_id).first()
    return jsonify({
        "id": user.id,
        "email": user.email
    }) 

@app.route("/register", methods=["POST"])
def register_user():
    data = request.get_json()
    email = data["email"]
    password = data["password"]

    user_exists = User.query.filter_by(email=email).first()

    if user_exists:
        return jsonify({"error": "User already exists"}), 409

    hashed_password = bcrypt.generate_password_hash(password).decode('utf-8')
    new_user = User(email=email, password=hashed_password)
    db.session.add(new_user)
    db.session.commit()
    
    session["user_id"] = new_user.id

    return jsonify({
        "id": new_user.id,
        "email": new_user.email
    })

@app.route("/login", methods=["POST"])
def login_user():
    data = request.get_json()
    email = data["email"]
    password = data["password"]

    user = User.query.filter_by(email=email).first()

    if user is None:
        return jsonify({"error": "Login details are incorrect"}), 401

    if not bcrypt.check_password_hash(user.password, password):
        return jsonify({"error": "Unauthorized"}), 401
    
    session["user_id"] = user.id

    return jsonify({
        "id": user.id,
        "email": user.email
    })

@app.route("/logout", methods=["POST"])
def logout_user():
    session.pop("user_id", None)
    return jsonify({"message": "Successfully logged out"})

**Comment by shaikmynuddin** [comment]
I want Mainly tech stacks of Application So it would be easy to integrate and I didn't received any request yet in the discord can paste link of your channel

**Comment by HomerusJa** [comment]
We are going for a FastAPI and SQLAlchemy tech stack in the backend. If you have any questions regarding the targeted architecture/the libraries used, feel free to ask. Can you prepend the routes with /auth?

**Comment by shaikmynuddin** [comment]
> We are going for a FastAPI and SQLAlchemy tech stack in the backend. If you have any questions regarding the targeted architecture/the libraries used, feel free to ask. Can you prepend the routes with /auth?

frontend
